src/components/ToolButton.tsx

```diff
@@ -41,20 +43,39 @@ const ToolButton: React.FC<ToolButtonProps> = ({
   onClick,
   icon,
   tooltipText,
+  toggle = false,
 }) => {
+  const [isToggled, setIsToggled] = useState(false);
+  const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
+    if (toggle) {
+      setIsToggled(!isToggled);
+    }
+    onClick(event);
+  };
+
   const iconComp = tooltipText ? (
     <Tooltip arrow title={tooltipText}>
       {icon}
     </Tooltip>
   ) : (
     icon
   );
+
   return (
     <IconButton
       className={className}
       disabled={disabled}
-      onClick={onClick}
+      onClick={handleClick}
       size="small"
+      sx={{
+        ...(toggle && {
+          border: "1px solid rgba(0, 0, 0, 0.12)",
+          borderRadius: "4px",
+          padding: "2.4px",
+          color: isToggled ? "rgba(0, 0, 0, 0.87)" : "rgba(0, 0, 0, 0.54)",
+          backgroundColor: isToggled ? "rgba(0, 0, 0, 0.08)" : "transparent",
+        }),
+      }}
```

Comment:
Use MUI's `ToggleButton`  <https://mui.com/material-ui/react-toggle-button/>.
Why didn't you use it?